cli/src/update/unix.rs

```diff
@@ -59,14 +57,50 @@ impl Default for ApplicationUpdater {
     }
 }
 
-/// Produces the path to a temporary file on disk.
-fn tmp_path(filename: &str) -> Option<String> {
-    let tmp_loc = env::temp_dir();
-    let path = Path::new(&tmp_loc);
-    let tmp_path = path.join(filename);
-    match tmp_path.into_os_string().into_string() {
-        Ok(x) => Some(x),
-        Err(_) => None,
+/// Generic function for fetching data via HTTP GET.
+async fn http_get(url: &str) -> anyhow::Result<reqwest::Response> {
+    let client = Client::builder().user_agent("phylum-cli").build()?;
+    let response = client.get(url).send().await?;
+    Ok(response)
+}
+
+/// Download the specified Github asset. Returns a bytes object containing the contents of the asset.
+async fn download_github_asset(latest: &GithubReleaseAsset) -> anyhow::Result<bytes::Bytes> {
+    let r = http_get(latest.browser_download_url.as_str()).await?;
+    Ok(r.bytes().await?)
+}
+
+const SUPPORTED_PLATFORMS: &[&str] = &[
+    "x86_64-unknown-linux-musl",
+    "x86_64-apple-darwin",
+    "aarch64-apple-darwin",
+];
+
+/// Determine the current platform. Error if unsupported
```

Comment:
```suggestion
/// Determine the current platform. Error if unsupported.
```

I'm not familiar with all the code conventions yet, but seems like trailing full stop is the norm.